webroot/src/components/Users/UserRow/UserRow.ts

```diff
@@ -0,0 +1,184 @@
+
+    //
+    // Data
+    //
+    lastSortSelectOption = '';
+    lastSortSelectDirection = '';
+    isRowExpanded = false;
+
+    //
+    // Computed
+    //
+    get userStore() {
+        return this.$store.state.user;
+    }
+
+    get currentCompactType() {
+        return this.userStore.currentCompact?.type;
+    }
+
+    get sortingStore() {
+        return this.$store.state.sorting;
+    }
+
+    get sortingStoreOption(): any {
+        return this.sortingStore.sortingMap[this.listId]?.option;
+    }
+
+    get sortingStoreDirection(): SortDirection {
+        return this.sortingStore.sortingMap[this.listId]?.direction;
+    }
+
+    get sortOptionNames(): Array<string> {
+        const names: Array<string> = [];
+
+        this.sortOptions?.forEach((option) => {
+            if (option.value) {
+                names.push(option.value);
+            }
+        });
+
+        return names;
+    }
+
+    get isAccountStatusPending(): boolean {
+        return this.item?.accountStatus === 'pending';
+    }
+
+    get isAccountStatusEmphasis(): boolean {
+        return this.isAccountStatusPending;
+    }
+
+    get shouldAllowResendInvite(): boolean {
+        return this.isAccountStatusPending;
+    }
+
+    //
+    // Methods
+    //
+    isSortOptionEnabled(optionName: string): boolean {
+        return Boolean(this.isHeaderRow && this.sortOptionNames.includes(optionName));
+    }
+
+    isSortOptionSelected(optionName: string): boolean {
+        return Boolean(this.isHeaderRow && this.sortingStoreOption === optionName);
+    }
+
+    sortOptionDirection(optionName: string): SortDirection {
+        const isOptionSelected = this.isSortOptionSelected(optionName);
+        let optionDirection = SortDirection.asc;
+
+        if (isOptionSelected) {
+            optionDirection = this.sortingStoreDirection;
+        }
+
+        return optionDirection;
+    }
+
+    isSortOptionAscending(optionName: string): boolean {
+        return Boolean(this.sortOptionDirection(optionName) === SortDirection.asc);
+    }
+
+    isSortOptionDescending(optionName: string): boolean {
+        return Boolean(this.sortOptionDirection(optionName) === SortDirection.desc);
+    }
+
+    async handleSortSelect(optionName: string): Promise<void> {
+        await this.handleSortChangeDirection(optionName);
+        await this.handleSortChangeOption(optionName);
+
+        if (this.sortChange) {
+            this.sortChange(this.sortingStoreOption, this.sortingStoreDirection === SortDirection.asc);
+        }
+    }
+
+    async handleSortChangeOption(optionName: string, isExternal = false): Promise<void> {
+        const sortingId = this.listId;
+
+        if (optionName !== this.sortingStoreOption) {
+            if (isExternal) {
+                if (this.lastSortSelectOption !== this.sortingStoreOption) {
+                    this.lastSortSelectOption = this.sortingStoreOption;
+                } else {
+                    // Continue
+                }
```

Comment:
Why can't you just set `lastSortSelectOption` without an if statement? 

Any reason to not just eliminate the else clause?

Reply:
Good catch. Updated.